packages/gestalt/src/SideNavigation.js

```diff
@@ -48,6 +41,18 @@ export type Props = {
    * Title for mobile navigation.
    */
   mobileTitle?: string,
+  /**
+   * When passed SideNavigation becomes a collapsible controlled component. If not passed, it stays non-collapsible. See the [collapsible variant](https://gestalt.pinterest.systems/web/sidenavigation#Collapsible) to learn more. This functionality is not supported in mobile.
+   */
+  collapsed?: boolean,
+  /**
+   * Callback fired when the SideNavigation is collapsible and the collapse/expand button is clicked. This functionality is not supported in mobile.
+   */
+  onCollapse?: (boolean) => void,
+  /**
+   * Callback fired when the *collapsed* SideNavigation is hovered and opened in preview mode. This functionality is not supported in mobile.
+   */
+  onPreview?: (boolean) => void,
```

Comment:
Another option could be to follow how we usually we've been doing this events, with named args

![Screenshot by Dropbox Capture](https://github.com/pinterest/gestalt/assets/10593890/45b792ee-b9c1-4d47-ab51-1837c1c5c3c9)


`onExpand?: ({ expanded: boolean, type: "default" | "preview" }) => void 
`

Said that, I think preview should be uncontrolled as we are delaying the collapse when on preview. If we had to support controlled previews... we would need to differenciate collapsed from previewed right? To add a different classname with the transition.

I think it's fine as it is. If we get requests for controlled we can change it in the future.

I guess the question is do we want the preview behavior as default? If we add a preview state prop that behaves similarly to collapsed then it could not be implemented and just be optional.

What's the actual use case of `onPreview`? Because I think the intention is to communicate state to header and footer right? Could all be just onCollapse (or onExpand if we change it?) Not sure the difference of this prop...

Reply:
@AlbertCarreras Preview is not only for communicating for the header and footer. Sorry if my comment misled you. Besides the header and footer, there are ellipsis items with contain multiple items, group items which need to be expanded when clicked. Those require sidenav to be in its normal width so they are displayed properly. That's why we need preview state.
I would suggest you to check out the Figma handoff to better understand the preview state.